runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2816,3 +2833,165 @@ JITaaSHelpers::getRemoteROMClass(J9Class *clazz, JITaaS::J9ServerStream *stream,
+      {
+      auto romClass = romClassFromString(std::get<0>(classInfoTuple), TR::comp()->trMemory()->trPersistentMemory());
+      JITaaSHelpers::cacheRemoteROMClass(clientSessionData, clazz, romClass, &classInfoTuple, classInfo);
+      JITaaSHelpers::getROMClassData(classInfo, dataType1, data1);
+      JITaaSHelpers::getROMClassData(classInfo, dataType2, data2);
+      }
+   else
+      {
+      JITaaSHelpers::getROMClassData(it->second, dataType1, data1);
+      JITaaSHelpers::getROMClassData(it->second, dataType2, data2);
+      }
+   return true;
+   }
+
+void
+JITaaSHelpers::getROMClassData(const ClientSessionData::ClassInfo &classInfo, ClassInfoDataType dataType, void *data)
+   {
+   switch (dataType)
+      {
+      case CLASSINFO_ROMCLASS_MODIFIERS :
+         {
+         *(uint32_t *)data = classInfo.romClass->modifiers;
+         }
+         break;
+      case CLASSINFO_ROMCLASS_EXTRAMODIFIERS :
+         {
+         *(uint32_t *)data = classInfo.romClass->extraModifiers;
+         }
+         break;
+      case CLASSINFO_BASE_COMPONENT_CLASS :
+         {
+         *(TR_OpaqueClassBlock **)data = classInfo.baseComponentClass;
+         }
+         break;
+      case CLASSINFO_NUMBER_DIMENSIONS :
+         {
+         *(int32_t *)data = classInfo.numDimensions;
+         }
+         break;
+      case CLASSINFO_PARENT_CLASS :
+         {
+         *(TR_OpaqueClassBlock **)data = classInfo.parentClass;
+         }
+         break;
+      case CLASSINFO_CLASS_HAS_FINAL_FIELDS :
+         {
+         *(bool *)data = classInfo.classHasFinalFields;
+         }
+         break;
+      case CLASSINFO_CLASS_DEPTH_AND_FLAGS :
+         {
+         *(uintptrj_t *)data = classInfo.classDepthAndFlags;
+         }
+         break;
+      case CLASSINFO_CLASS_INITIALIZED :
+         {
+         *(bool *)data = classInfo.classInitialized;
+         }
+         break;
+      case CLASSINFO_BYTE_OFFSET_TO_LOCKWORD :
+         {
+         *(uint32_t *)data = classInfo.byteOffsetToLockword;
+         }
+         break;
+      case CLASSINFO_LEAF_COMPONENT_CLASS :
+         {
+         *(TR_OpaqueClassBlock **)data = classInfo.leafComponentClass;
+         }
+         break;
+      case CLASSINFO_CLASS_LOADER :
+         {
+         *(void **)data = classInfo.classLoader;
+         }
+         break;
+      case CLASSINFO_HOST_CLASS :
+         {
+         *(TR_OpaqueClassBlock **)data = classInfo.hostClass;
+         }
+         break;
+      case CLASSINFO_COMPONENT_CLASS :
+         {
+         *(TR_OpaqueClassBlock **)data = classInfo.componentClass;
+         }
+         break;
+      case CLASSINFO_ARRAY_CLASS :
+         {
+         *(TR_OpaqueClassBlock **)data = classInfo.arrayClass;
+         }
+         break;
+      case CLASSINFO_TOTAL_INSTANCE_SIZE :
+         {
+         *(uintptrj_t *)data = classInfo.totalInstanceSize;
+         }
+         break;
+      default :
+         break;
```

Comment:
I suggest an ASSERT for the default case. This will catch errors when we add new fields.